out_writeapi.go

```diff
@@ -145,7 +147,9 @@ func checkResponses(curr_ctx context.Context, currQueuePointer *[]*managedwriter
 			select {
 			case <-queueHead.Ready():
 				recvOffset, err := queueHead.GetResult(curr_ctx)
+				(*currMutex).Lock()
```

Comment:
Under flush, we should check the queue head, get the result and update the queue all under the lock. That's because if another thread called flush, both threads might see that the head was ready, and then both proceed to grab the lock and remove the head element. This would make us remove 2 elements rather than 1.

Reply:
In that case, could I just put the mutex around the call to checkResponses in flush (line 293)? Would that be better than putting it around those specific areas in checkResponses?